source/constructs/api/data_source/glue_database_detector.py

```diff
@@ -38,28 +41,26 @@ async def detect_glue_database_connection(session: Session, aws_account_id: str)
             aws_session_token=credentials['SessionToken'],
             region_name=region
         )
-        glue_database_list.append(client.get_databases()['DatabaseList'])
-        logger.info("detect_glue_database")
+        glue_database_list = client.get_databases()['DatabaseList']
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `get_databases` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)